pybop/models/base_model.py

```diff
@@ -504,44 +505,50 @@ def predict(
             if PyBaMM models are not supported by the current simulation method.
 
         """
-        inputs = self.parameters.verify(inputs)
-
-        if not self.pybamm_model._built:
-            self.pybamm_model.build_model()
+        if self._unprocessed_model is None:
+            raise ValueError(
+                "The predict method currently only supports PyBaMM models."
+            )
+        elif not self._unprocessed_model._built:
+            self._unprocessed_model.build_model()
 
         parameter_set = parameter_set or self._unprocessed_parameter_set
         if inputs is not None:
+            inputs = self.parameters.verify(inputs)
             parameter_set.update(inputs)
 
+        if init_soc is not None and "Initial SoC" in parameter_set.keys():
+            parameter_set["Initial SoC"] = init_soc
```

Comment:
This line is for ECMs which have an "Initial SoC" parameter. This line is used for both passing the new "Initial SoC" to PyBaMM's Simulation class and also to update the default parameter set for use in future calls to simulate or predict.

Reply:
Thanks, in which case, I think it would be more robust to have an `isinstance()` condition instead of the string comparison. It's possible for a user to add their own "Initial SoC" key to a `parameter_set` for use in the physics-based models, which would currently break the `pybamm.Simulation.solve()` initial_soc calculation.

Reply:
How do I get around the circular import?

Reply:
One way would be to check if the `_model` property is an instance of `pybamm.Thevenin`.